src/mongo/delegates/Views.class.php

```diff
@@ -396,58 +406,16 @@ public function generateView($viewId,$resource=null,$context=null,$queueName=nul
             $t = new \Tripod\Timer();
             $t->start();
 
-            $from = $this->getFromCollectionForViewSpec($viewSpec);
+            $from = $viewSpec['from'];
             $collection = $this->config->getCollectionForView($this->storeName, $viewId);
 
+            $this->ensureIndices($viewSpec, $collection);
```

Comment:
I am all for optimizing ways to do this.  My comment wasn't necessarily a defense of the status quo, more just a rationale that we hadn't come up with/prioritized anything better.

Historically (and, I imagine, as you're currently seeing), there has been a lot of "HERE BE DRAGONS" when it comes to the composites functionality in Tripod.  A lot of reticence to pull at too many strings because, while we have a remit to fix broken windows, sometimes those windows are looking into the howling void of madness.  It's also a tremendous time sink not to mention a white-knuckle "how good is our test coverage _really_?" when you get into it.